packages/edit-site/src/components/save-panel/index.js

```diff
@@ -7,17 +7,72 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { Button, Modal } from '@wordpress/components';
-import { EntitiesSavedStates } from '@wordpress/editor';
+import {
+	EntitiesSavedStates,
+	EntitiesSavedStatesExtensible,
+	useEntitiesSavedStatesIsDirty,
+} from '@wordpress/editor';
 import { useDispatch, useSelect } from '@wordpress/data';
-import { __ } from '@wordpress/i18n';
+import { __, sprintf } from '@wordpress/i18n';
 import { NavigableRegion } from '@wordpress/interface';
+import { store as coreStore } from '@wordpress/core-data';
 
 /**
  * Internal dependencies
  */
 import { store as editSiteStore } from '../../store';
 import { unlock } from '../../private-apis';
 import { useActivateTheme } from '../../utils/use-activate-theme';
+import {
+	currentlyPreviewingTheme,
+	isPreviewingTheme,
+} from '../../utils/is-previewing-theme';
+
+const _EntitiesSavedStates = ( { onClose } ) => {
+	const { getTheme } = useSelect( coreStore );
+	const theme = getTheme( currentlyPreviewingTheme() );
+	const activateTheme = useActivateTheme();
+	const isDirtyProps = useEntitiesSavedStatesIsDirty();
+	const saveEnabled = isPreviewingTheme() || isDirtyProps.isDirty;
+
+	let activateSaveLabel;
+	if ( isPreviewingTheme() && isDirtyProps.isDirty ) {
+		activateSaveLabel = __( 'Activate & Save' );
+	} else if ( isPreviewingTheme() ) {
+		activateSaveLabel = __( 'Activate' );
+	} else {
+		activateSaveLabel = undefined;
+	}
+
+	const additionalPrompt = (
+		<p>
+			{ sprintf(
+				'Saving your changes will change your active theme to  %1$s.',
+				theme?.name?.rendered
+			) }
+		</p>
+	);
+
+	const onSave = async ( values ) => {
+		await activateTheme();
+		return values;
+	};
+
+	return window?.__experimentalEnableThemePreviews ? (
```

Comment:
I thought the purpose of this PR is to remove the experiment. Do we still need this conditional?

Reply:
Thanks for your comment!
I have removed `window.__experimentalEnableThemePreviews` here.
https://github.com/WordPress/gutenberg/pull/50863/commits/415c7fb627c73f10a5fb48445a6189f9d93b49ab

Also, I have tried to remove the option for experiment here, does it make sense to you? @scruffian 
https://github.com/WordPress/gutenberg/pull/50863/commits/1f8961aa2b89552fc6f6b6e521dc696923c85547

Reply:
See https://github.com/WordPress/gutenberg/pull/50863#discussion_r1205159565